fv3core/stencils/del2cubed.py

```diff
@@ -73,47 +70,98 @@ def corner_fill(q: FloatField):
             q = q[0, -1, 0]
 
 
-def compute(qdel: FloatField, nmax: int, cd: float, km: int):
-    grid = spec.grid
-    origin = (grid.isd, grid.jsd, 0)
-
-    # Construct some necessary temporary storage objects
-    fx = utils.make_storage_from_shape(
-        qdel.shape, origin=origin, cache_key="del2cubed_fx"
-    )
-    fy = utils.make_storage_from_shape(
-        qdel.shape, origin=origin, cache_key="del2cubed_fy"
-    )
-
-    # set up the temporal loop
-    ntimes = min(3, nmax)
-    for n in range(1, ntimes + 1):
-        nt = ntimes - n
-        origin = (grid.is_ - nt, grid.js - nt, 0)
+class HyperdiffusionDamping:
+    """
+    Fortran name is del2cubed
+    """
+
+    def __init__(self, grid, qdel: FloatField, km: int):
```

Comment:
I'm glad you are leaving km as an argument -- one might remove it since the current python code only calls it with one possible value, but there are namelist flags in the fortran code that trigger using different values. you may want to assign self._km = km so you can use it in the call method.

Reply:
actually that's incorrect, km is only ever grid.npz so can be removed :)

Reply:
km is now removed and replaced with `grid.npz`